boot/bootutil/src/loader.c

```diff
@@ -2149,6 +2156,7 @@ context_boot_go(struct boot_loader_state *state, struct boot_rsp *rsp)
              */
             rc = boot_read_image_headers(state, false, &bs);
             if (rc != 0) {
+                FIH_SET(fih_rc, FIH_FAILURE);
```

Comment:
seems redundant with lines 2224-2226. Is it on purpose?  The same applies to all changes below where `FIH_SET` is used

Reply:
Previously the construction where `rc` was set to -1 was weak. Setting `fih_rc` to FIH_FAILURE solved the problem. But you are right there is some redundacy here. We could replace FIH_SET with a simple affectation or remove the last test to save some clock cycles.